File: projects/Backend/TheRocket/TheRocket/Entities/Users/AppUser.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Identity;
using TheRocket.Entities;
using TheRocket.Entities.Users;

namespace TheRocket.Entities.Users
{
    public class AppUser:IdentityUser
    {
        public AppUser()
        {
            Addresses=new();
            PhoneNumbers=new();
            Locations=new();
        }
        
        public virtual Seller? Seller { get; set; }

        public virtual Buyer? Buyer { get; set; }
        public virtual Admin? Admin { get; set; }

        public virtual List<Phone> PhoneNumbers { get; set; }

        public virtual List<Address> Addresses { get; set; }

        public virtual List<Location>? Locations{get; set;}
        
    }


   
}

File: projects/Frontend/client/src/app/shared/models/appUser.ts
import { IAddress } from "./IAddress";
import { IAdmin } from "./admin";
import { ILocation } from "./location";
import { IPhoneNumber } from "./phoneNumber";
import { Seller } from "./seller";
import { Buyer } from "./buyer";

export class AppUser {
    id?: string;
    userName?: string;
    email?: string;
    password?: string;
    confirmPassword?: string;
    seller?: Seller;
    buyer?: Buyer;
    admin?: IAdmin;
    phoneNumbers?: IPhoneNumber[];
    addresses?: IAddress[];
    locations?: ILocation[];
    accountType?: number;

    constructor() {
    }
}